documents.go

```diff
@@ -133,3 +133,75 @@ func (cl *DocumentsCreateClient) Do(ctx context.Context) (*Document, error) {
 
 	return success.Data, nil
 }
+
+// documentsUpdateParams represents the Outline Documents.update parameters
+type documentsUpdateParams struct {
+	Id      DocumentID `json:"id"`
+	Title   string     `json:"title,omitempty"`
+	Text    string     `json:"text,omitempty"`
+	Append  bool       `json:"append,omitempty"`
+	Publish bool       `json:"publish,omitempty"`
+	Done    bool       `json:"done,omitempty"`
+}
+
+// DocumentsUpdateClient is a client for updating a single document.
+type DocumentsUpdateClient struct {
+	sl     *sling.Sling
+	params documentsUpdateParams
+}
+
+func newDocumentsUpdateClient(sl *sling.Sling, params documentsUpdateParams) *DocumentsUpdateClient {
+	copy := sl.New()
+
+	return &DocumentsUpdateClient{sl: copy, params: params}
+}
```

Comment:
[Here](https://github.com/ioki-mobility/go-outline/pull/29#discussion_r1282424980) I got told from @rsjethani that we should create the `Params` in this function and put the required (API) parameters as (function) parameters. In this case, the `DocumentId`.

I guess we should do the same here.

Reply:
Very good, thanks - I'll fix that :+1:

Reply:
I've also changed that for documents create in a separate commit, but we can also separate it into it's own PR